integrations/hanlp/src/haystack_integrations/components/preprocessors/hanlp/chinese_document_splitter.py

```diff
@@ -0,0 +1,499 @@
+    ):
+        """
+        Initialize the ChineseDocumentSplitter component.
+
+        :param split_by: The unit for splitting your documents. Choose from:
+            - `word` for splitting by spaces (" ")
+            - `period` for splitting by periods (".")
+            - `page` for splitting by form feed ("\\f")
+            - `passage` for splitting by double line breaks ("\\n\\n")
+            - `line` for splitting each line ("\\n")
+            - `sentence` for splitting by HanLP sentence tokenizer
+
+        :param split_length: The maximum number of units in each split.
+        :param split_overlap: The number of overlapping units for each split.
+        :param split_threshold: The minimum number of units per split. If a split has fewer units
+            than the threshold, it's attached to the previous split.
+        :param respect_sentence_boundary: Choose whether to respect sentence boundaries when splitting by "word".
+            If True, uses HanLP to detect sentence boundaries, ensuring splits occur only between sentences.
+        :param splitting_function: Necessary when `split_by` is set to "function".
+            This is a function which must accept a single `str` as input and return a `list` of `str` as output,
+            representing the chunks after splitting.
+        :param granularity: The granularity of Chinese word segmentation, either 'coarse' or 'fine'.
+
+        :raises ValueError: If the granularity is not 'coarse' or 'fine'.
+        """
+        self.split_by = split_by
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_threshold = split_threshold
+        self.respect_sentence_boundary = respect_sentence_boundary
+        self.splitting_function = splitting_function
+        self.granularity = granularity
+
+        if granularity not in {"coarse", "fine"}:
+            msg = f"Invalid granularity '{granularity}'. Choose either 'coarse' or 'fine'."
+            raise ValueError(msg)
+
+    def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
+        """
+        Split documents into smaller chunks.
+
+        :param documents: The documents to split.
+        :return: A dictionary containing the split documents.
+
+        :raises RuntimeError: If the Chinese word segmentation model is not loaded.
+        """
+        if self.split_sent is None:
+            msg = "The Chinese word segmentation model is not loaded. Please run 'warm_up()' before calling 'run()'."
+            raise RuntimeError(msg)
+
+        split_docs = []
+        for doc in documents:
+            split_docs.extend(self._split_document(doc))
+        return {"documents": split_docs}
+
+    def warm_up(self) -> None:
+        """Warm up the component by loading the necessary models."""
+        if self.granularity == "coarse":
+            self.chinese_tokenizer = hanlp.load(hanlp.pretrained.tok.COARSE_ELECTRA_SMALL_ZH)
```

Comment:
Maybe we can add a try/except around the `hanlp.load` operation so that when it can't load a model fails safely.